packages/flutter_tools/lib/src/commands/create.dart

```diff
@@ -548,14 +640,24 @@ To edit platform code in an IDE see https://flutter.dev/developing-packages/#edi
 
     if (boolArg('pub')) {
       await pub.get(context: PubContext.create, directory: directory.path, offline: boolArg('offline'));
-      await project.ensureReadyForPlatformSpecificTooling(checkProjects: false);
+      await project.ensureReadyForPlatformSpecificTooling(checkProjects: pluginExampleApp);
+    }
+    if (templateContext['android'] == true) {
+      gradle.updateLocalProperties(project: project, requireAndroidSdk: false);
     }
-
-    gradle.updateLocalProperties(project: project, requireAndroidSdk: false);
-
     return generatedCount;
   }
 
+  void _updateTemplateContextWithPlatforms(Map<String, dynamic> context, List<String> platforms) {
```

Comment:
This is confusing, everything gets set to false by default, then everything get sets to true, then some are conditional set to false.